core/src/main/java/cucumber/runtime/formatter/HTMLFormatter.java

```diff
@@ -193,14 +193,22 @@ private void handleEmbed(EmbedEvent event) {
         String mimeType = event.mimeType;
         if(mimeType.startsWith("text/")) {
             // just pass straight to the formatter to output in the html
-            jsFunctionCall("embedding", mimeType, new String(event.data));
+            if (event.name != null) {
+                jsFunctionCall("embedding", mimeType, new String(event.data), event.name);
+            } else {
+                jsFunctionCall("embedding", mimeType, new String(event.data));
+            }
         } else {
             // Creating a file instead of using data urls to not clutter the js file
             String extension = MIME_TYPES_EXTENSIONS.get(mimeType);
             if (extension != null) {
                 StringBuilder fileName = new StringBuilder("embedded").append(embeddedIndex++).append(".").append(extension);
                 writeBytesToURL(event.data, toUrl(fileName.toString()));
-                jsFunctionCall("embedding", mimeType, fileName);
+                if (event.name != null) {
+                    jsFunctionCall("embedding", mimeType, fileName, event.name);
+                } else {
+                    jsFunctionCall("embedding", mimeType, fileName);
```

Comment:
Would it be possible to fix `jsFunctionCall` so this if statement isn't needed?

Reply:
Or I guess fix the `embedding` that it calls.

Reply:
I removed this 'if' statement.